File: joao pedro/Problema0440/Problema0440.java
import java.util.Scanner;

public class Problema0440 {
    public static double funcao(double[] x, int i, int z) {

        if (z == 1) {
            return i;
        } else {
            return -1;
        }
    }

    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        /*
         * Seja X um vetor com N elementos inteiros. Escreva uma função que forneça como
         * resultado
         * a posição do primeiro valor negativo do arranjo X. Caso X não contenha nenhum
         * negativo, a função deve fornecer
         * como resultado o valor -1. Em seguida, crie um programa que leias os dados
         * necessários, chame a função e apresente o
         * valor retornado por ela.
         */
        int z = 0;
        int h = 0;
        int a = 0;

        System.out.println("Digite o tamanho do vetor x: ");
        int n = Integer.valueOf(input.nextLine());
        double[] x = new double[n];
        for (int i = 0; i < n; i++) {
            System.out.println("Digite o valor na posicao " + i + ": ");
            x[i] = Double.valueOf(input.nextLine());
            if (x[i] < 0 && z == 0) {
                funcao(x, i, z);
                z = 1;
                h = i;

            }

        }
        for (int j = 0; j < n; j++) {
            if (z == 0) {
                funcao(x, j, z);
                a = j;
            }
        }
        if (z == 1) {
            System.out.println(funcao(x, h, z));
        } else {
            System.out.println(funcao(x, a, z));
        }
    }

}

File: joao pedro/Problema0440/Problema0440.py
def funcao(x,i,z):
    if z==1:
        return i
    else:
        return -1


def main():
    print("Digite '1111' para parar")
    x=[]
    i=0
    z=0
    y=0
    a=0
    h=0
    while(y!=1111):
        print("Digite um valor para a posicao ",i,": ")
        y=float(input(""))
        x.insert(i,y)
        if x[i]<0 and z!=1:
            z=1
            h=i
            funcao(x,i,z)
        i+=1    
    for y in range (i):
        if z==0:
            a=y
            funcao(x,y,z)
    if(z==0):
        print(funcao(x,a,z))
    else:
        print(funcao(x,h,z))

main()

        


    